### PR TITLE
lib/posix-user: Add comments for posix-user

### PR DESCRIPTION
**Pull request by DeliaPavel**
### Prerequisite checklist

 - [ ] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [ ] Tested your changes against relevant architectures and platforms;
 - [ ] Ran the [`checkpatch.uk`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.uk) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.

### Base target

 - Architecture(s): [e.g. `x86_64` or N/A]
 - Platform(s): [e.g. `kvm`, `xen` or N/A]
 - Application(s): [e.g. `app-python3` or N/A]

### CODE REVIEW

**Comment by PAN740623** [comment]
错误报告 阻止

**Comment by PAN740623** [comment]
url:

File: lib/posix-user/musl-imported/include/grp.h [hunk 1]
```diff
@@ -15,33 +15,161 @@ extern "C" {
...
+ * information.
+ * This is part of the POSIX documentation, refer to that for more details.
+ *
+ * @param gid
+ *   Group id to search by
+ * @return
+ *   Pointer to the group with group id `gid`.
+ */
 struct group  *getgrgid(gid_t);
+/**
```
Comment (StefanJum) [inline lib/posix-user/musl-imported/include/grp.h:45]:
```suggestion

/**
```

File: lib/posix-user/musl-imported/include/grp.h [hunk 2]
```diff
@@ -15,33 +15,161 @@ extern "C" {
...
+ *	 Length of buffer `buf`
+ * @param result
+ *   Pointer to a location in which a pointer to the updated `group` structure
+ *   is stored. If an error occurs or if the requested entry cannot be found,
+ *   a NULL pointer is stored in this location
+ * @return
+ *   0 on success, `ERANGE` if `buflen` is not large enough.
+ */
 int getgrgid_r(gid_t, struct group *, char *, size_t, struct group **);
+/**
```
Comment (StefanJum) [inline lib/posix-user/musl-imported/include/grp.h:81]:
```suggestion

/**
```

File: lib/posix-user/musl-imported/include/grp.h [hunk 3]
```diff
@@ -15,33 +15,161 @@ extern "C" {
...
+ *   Pointer to a location in which a pointer to the updated `group` structure
+ *   is stored. If an error occurs or if the requested entry cannot be found,
+ *   a NULL pointer is stored in this location
+ * @return
+ *   0 on success, `ERANGE` if `buflen` is not large enough.
+ */
 int getgrnam_r(const char *, struct group *, char *, size_t, struct group **);
 
 #if defined(_XOPEN_SOURCE) || defined(_GNU_SOURCE) || defined(_BSD_SOURCE)
+/**
```
Comment (StefanJum) [inline lib/posix-user/musl-imported/include/grp.h:106]:
```suggestion

/**
```

File: lib/posix-user/musl-imported/include/grp.h [hunk 4]
```diff
@@ -15,33 +15,161 @@ extern "C" {
...
+ * The first time this function is called, it returns the first entry.
+ * Afterward, it returns successive entries.
+ * This function allows iterating over all the groups from the array.
+ * This is part of the POSIX documentation, refer to that for more details.
+ *
+ * @return
+ *	 The current group entry.
+ */
 struct group  *getgrent(void);
+/**
```
Comment (StefanJum) [inline lib/posix-user/musl-imported/include/grp.h:117]:
```suggestion

/**
```

File: lib/posix-user/musl-imported/include/grp.h [hunk 5]
```diff
@@ -15,33 +15,161 @@ extern "C" {
...
+ * This function is normally used to close the group database,
+ * but since, in this case, this is just an array, it does nothing.
+ * This is part of the POSIX documentation, refer to that for more details.
+ */
 void           endgrent(void);
+/**
```
Comment (StefanJum) [inline lib/posix-user/musl-imported/include/grp.h:124]:
```suggestion

/**
```

File: lib/posix-user/musl-imported/include/grp.h [hunk 6]
```diff
@@ -15,33 +15,161 @@ extern "C" {
...
+ * Sets the groups iterator at the beginning of the `groups` array.
+ * This is part of the POSIX documentation, refer to that for more details.
+ */
 void           setgrent(void);
 #endif
 
 #ifdef _GNU_SOURCE
+/**
```
Comment (StefanJum) [inline lib/posix-user/musl-imported/include/grp.h:133]:
```suggestion

/**
```

File: lib/posix-user/musl-imported/include/grp.h [hunk 7]
```diff
@@ -15,33 +15,161 @@ extern "C" {
...
+ * This is part of the POSIX documentation, refer to that for more details.
+ */
 struct group  *fgetgrent(FILE *);
+/**
```
Comment (StefanJum) [inline lib/posix-user/musl-imported/include/grp.h:138]:
```suggestion

/**
```

File: lib/posix-user/musl-imported/include/grp.h [hunk 8]
```diff
@@ -15,33 +15,161 @@ extern "C" {
...
+ * This is part of the POSIX documentation, refer to that for more details.
+ */
 int putgrent(const struct group *, FILE *);
 #endif
 
 #if defined(_GNU_SOURCE) || defined(_BSD_SOURCE)
+/**
```
Comment (StefanJum) [inline lib/posix-user/musl-imported/include/grp.h:146]:
```suggestion

/**
```

File: lib/posix-user/musl-imported/include/grp.h [hunk 9]
```diff
@@ -15,33 +15,161 @@ extern "C" {
...
+ * @param groups
+ *   Array of groups the user belongs to
+ * @param ngroups
+ *   Maximum number of groups to return in the list at input, number of groups
+ *   the user belogs to at output
+ * @return
+ *	 The number of groups the user belongs to, or -1 on error.
+ */
 int getgrouplist(const char *, gid_t, gid_t *, int *);
+/**
```
Comment (StefanJum) [inline lib/posix-user/musl-imported/include/grp.h:165]:
```suggestion

/**
```

File: lib/posix-user/musl-imported/include/grp.h [hunk 10]
```diff
@@ -15,33 +15,161 @@ extern "C" {
...
+ *
+ * @param size
+ *   Size of the `list` buffer
+ * @param list
+ *   List of supplementary group IDs
+ * @return
+ *	 0 on success, -1 on error.
+ */
 int setgroups(size_t, const gid_t *);
+/**
```
Comment (StefanJum) [inline lib/posix-user/musl-imported/include/grp.h:180]:
```suggestion

/**
```

File: lib/posix-user/user.h [hunk 11]
```diff
@@ -43,9 +43,33 @@
...
 
+/**
+ * Initializes the `passwds` list and inserts in it the first entry.
+ * This entry contains the default values defined above, such that
+ * name => `UK_DEFAULT_USER`, passwd => `UK_DEFAULT_PASS`,
+ * uid => `UK_DEFAULT_UID`, gid => `UK_DEFAULT_GID`,
+ * gecos => `UK_DEFAULT_USER`, dir = "/", shell = "".
+ */
 void pu_init_passwds(void);
+/**
```
Comment (StefanJum) [inline lib/posix-user/user.h:54]:
```suggestion

/**
```

**Review by StefanJum — CHANGES_REQUESTED** [review]
Few comments @DeliaPavel, looks good overall.

File: lib/posix-user/musl-imported/include/grp.h [hunk 12]
```diff
@@ -15,33 +15,171 @@ extern "C" {
...
 
+/**
+ * Like `getgrgid`, but stores the retrieved `group`
+ * structure in the space pointed to by `grp`.
+ * This is part of the POSIX documentation, refer to that for more details.
+ *
+ * @param gid
+ *   Group id to search by
+ * @param grp
+ *	 Group structure in which the group with the `gid` group id is stored
```
Comment (StefanJum) [inline lib/posix-user/musl-imported/include/grp.h]:
```suggestion
 *   Group structure in which the group with the `gid` group id is stored
```
Do not use tabs within comment blocks.

File: lib/posix-user/musl-imported/include/grp.h [hunk 13]
```diff
@@ -15,33 +15,171 @@ extern "C" {
...
+ * @param buf
+ *   Buffer which holds the string fields pointed to by the members of the
+ *   `group` structure
+ * @param buflen
+ *	 Length of buffer `buf`
```
Comment (StefanJum) [inline lib/posix-user/musl-imported/include/grp.h]:
```suggestion
 *   Length of buffer `buf`
```

File: lib/posix-user/musl-imported/include/grp.h [hunk 14]
```diff
@@ -15,33 +15,171 @@ extern "C" {
...
+
+/**
+ * Like `getgrnam`, but stores the retrieved `group` structure
+ * in the space pointed to by `grp`.
+ * This is part of the POSIX documentation, refer to that for more details.
+ *
+ * @param name
+ *   Group name to search by
+ * @param grp
+ *	 Group structure in which the group with the `name` group name is stored
```
Comment (StefanJum) [inline lib/posix-user/musl-imported/include/grp.h]:
```suggestion
 *   Group structure in which the group with the `name` group name is stored
```

File: lib/posix-user/musl-imported/include/grp.h [hunk 15]
```diff
@@ -15,33 +15,171 @@ extern "C" {
...
+ * @param buf
+ *   Buffer which holds the string fields pointed to by the members of the
+ *   `group` structure
+ * @param buflen
+ *	 Length of buffer `buf`
```
Comment (StefanJum) [inline lib/posix-user/musl-imported/include/grp.h]:
```suggestion
 *   Length of buffer `buf`
```

File: lib/posix-user/musl-imported/include/grp.h [hunk 16]
```diff
@@ -15,33 +15,171 @@ extern "C" {
...
+ * @param group
+ *   Group which will be included in `groups`,
+ *   if it was not among the groups defined for `user`
+ * @param groups
+ *   Array of groups the user belongs to
+ * @param ngroups
+ *   Maximum number of groups to return in the list at input, number of groups
+ *   the user belogs to at output
+ * @return
+ *	 The number of groups the user belongs to, or -1 on error.
```
Comment (StefanJum) [inline lib/posix-user/musl-imported/include/grp.h]:
```suggestion
 *   The number of groups the user belongs to, or -1 on error.
```

File: lib/posix-user/musl-imported/include/grp.h [hunk 17]
```diff
@@ -15,33 +15,171 @@ extern "C" {
...
+ * group (the default group), so this function only tests if the caller tries
+ * to set any invalid groups.
+ * This is part of the POSIX documentation, refer to that for more details.
+ *
+ * @param size
+ *   Size of the `list` buffer
+ * @param list
+ *   List of supplementary group IDs
+ * @return
+ *	 0 on success, -1 on error.
```
Comment (StefanJum) [inline lib/posix-user/musl-imported/include/grp.h]:
```suggestion
 *   0 on success, -1 on error.
```

File: lib/posix-user/user.h [hunk 18]
```diff
@@ -43,9 +43,33 @@
...
 void pu_init_groups(void);
 
+/**
+ * Copies `src` string at the end of the `dest` string,
+ * also adding `\0` at the end.
+ *
+ * @param src
+ *   Source string
+ * @param dest
+ *	 Destination string
```
Comment (StefanJum) [inline lib/posix-user/user.h]:
```suggestion
 *   Destination string
```

**Review by StefanJum — CHANGES_REQUESTED** [review]
Almost done, see the comments @DeliaPavel

**Review by StefanJum — APPROVED** [review]
Thanks @DeliaPavel 
Approved-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Review by razvand — COMMENTED** [review]
Reviewed-by: Razvan Deaconescu <razvand@unikraft.io>

**Review by razvand — APPROVED** [review]
Reviewed-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request merged**